src/roles/matchmaker.py

```diff
@@ -33,39 +48,36 @@ def _set_lovers(target1, target2):
     target1.send(messages["matchmaker_target_notify"].format(target2))
     target2.send(messages["matchmaker_target_notify"].format(target1))
 
-def get_lovers(var: GameState):
+def get_all_lovers() -> list[set[User]]:
     lovers = []
-    pl = get_players(var)
-    for lover in LOVERS:
-        done = None
-        for i, lset in enumerate(lovers):
-            if lover in pl and lover in lset:
-                if done is not None: # plot twist! two clusters turn out to be linked!
-                    done.update(lset)
-                    for lvr in LOVERS[lover]:
-                        if lvr in pl:
-                            done.add(lvr)
-
-                    lset.clear()
-                    continue
-
-                for lvr in LOVERS[lover]:
-                    if lvr in pl:
-                        lset.add(lvr)
-                done = lset
-
-        if done is None and lover in pl:
-            lovers.append(set())
-            lovers[-1].add(lover)
-            for lvr in LOVERS[lover]:
-                if lvr in pl:
-                    lovers[-1].add(lvr)
-
-    while set() in lovers:
-        lovers.remove(set())
+    all_lovers = set(LOVERS.keys())
+    while all_lovers:
+        cur = all_lovers.pop()
+        visited = {cur}
```

Comment:
Should this call `get_lovers` for every lover? Could pass a `visited` kwarg so it communicates well; would make sure the logic is identical.